kobo/apps/openrosa/apps/api/viewsets/xform_submission_api.py

```diff
@@ -61,6 +63,49 @@ def create_instance_from_json(username, request):
     return safe_create_instance(username, xml_file, [], None, request=request)
 
 
+def extract_confirmation_message(xml_string: str) -> str | None:
+    """
+    Extracts the confirmation message from the XML string based on the
+    `kobo:submitMessage` attribute.
+    """
+    if isinstance(xml_string, str):
+        xml_string = xml_string.encode('utf-8')
+    parser = etree.XMLParser(recover=True)
+    root = etree.fromstring(xml_string, parser=parser)
+    namespaces = root.nsmap
+
+    # Extract the kobo:submitMessage attribute from the root element
+    try:
+        confirmation_message_xpath = root.xpath(
+            '@kobo:submitMessage', namespaces=namespaces
+        )
+    except etree.XPathEvalError:
+        return
+
+    if not confirmation_message_xpath:
+        return
+
+    # Blocked by kpi#5137, this block below works as-is but won't work
+    # when kpi#5137 is merged.
+    confirmation_message_xpath = confirmation_message_xpath[0].replace(
+        '/data', f'/{root.tag}'
+    ).strip()
+
+    try:
+        # Evaluate the XPath expression to find the message
+        confirmation_message_element = root.xpath(confirmation_message_xpath)
+    except etree.XPathEvalError as e:
+        logging.error(
+            'Failed to extract confirmation message: ' + str(e),
+            exc_info=True
+        )
+    else:
+        if confirmation_message_element:
+            return confirmation_message_element[0].text
+
+    return
```

Comment:
I took the liberty to make some changes to understand how it worked and to make it work.
The tests were passing because in your XML, `submitMessage` starts with the root tag. 

In real world, Kobo uses the `id_string` for the XForm which always changes and cannot be set in the `settings` for the XLS file (i.e. `attribute::kobo:submitMessage`) 

See https://chat.kobotoolbox.org/#narrow/stream/4-Kobo-Dev/topic/Customizable.20confirmation.20screen.20in.20Enketo/near/469885 , IMO this PR is blocked by kpi#5137. 

We should add a test to be sure everything work if no `attribute::kobo:submitMessage` is present.